packages/grafana-data/src/types/datasource.ts

```diff
@@ -218,6 +218,11 @@ export abstract class DataSourceApi<
    */
   getQueryDisplayText?(query: TQuery): string;
 
+  /**
+   * Convert a simple text string to query
+   */
+  getQueryFromDisplayText?(query: string): any;
+
   /**
```

Comment:
Datasources, that implement `getQueryDisplayText` method should implement `getQueryFromDisplayText` as well. This way we can translate query text to query.